api/src/libs/githubInvitationInit.ts

```diff
@@ -0,0 +1,162 @@
+import { createAppAuth } from '@octokit/auth'
+import { request } from '@octokit/request'
+import { getGithubPrivateKey } from '../config/githubConfig'
+import { logger } from '@bcgov/common-nodejs-utils';
+
+// cached value
+type APPInitailValue = {
+  initialized: number | null,
+  apps: object,
+  nonInstallatedApp: Function | null,
+}
+
+const  installationApps = <APPInitailValue>{
+  initialized: null,
+  apps: {},
+  nonInstallatedApp: null
+}
+
+/**
+ * getNonInstallationApp
+ * @returns a non installed github app
+ */
+export const getNonInstallationApp = () => {
+    logger.info('getNonInstallationApp')
+  // caches a non installed app
+  try{
+  if (!installationApps.nonInstallatedApp) {
+    const auth = createAppAuth({
+      appId: process.env.GITHUB_APP_ID ||  131752,
+      privateKey: getGithubPrivateKey(),
+      clientId: process.env.GITHUB_CLIENT_ID,
+      clientSecret: process.env.GITHUB_CLIENT_SECRET,
+    })
+
+    installationApps.nonInstallatedApp = request.defaults({
+      request: {
+        hook: auth.hook,
+      },
+      mediaType: {
+        previews: ['machine-man'],
+      },
+    })
+  }
+ } catch(e){
+    logger.error('I catch an error', e);
+    
+
+  }
+
+
+  return installationApps.nonInstallatedApp
+}
+
+const newAuthorizedApp = (installationId) => {
+
+  const app = createAppAuth({
+    appId: process.env.GITHUB_APP_ID || 131752,
+    privateKey: getGithubPrivateKey(),
+    clientId: process.env.GITHUB_CLIENT_ID,
+    clientSecret: process.env.GITHUB_CLIENT_SECRET,
+    installationId,
+  })
+
+  return {
+    initialized: Date.now(),
+    app,
+    id: installationId,
+    authenticatedRequest: request.defaults({
+      request: {
+        hook: app.hook,
+      },
+      mediaType: {
+        previews: ['machine-man'],
+      },
+    }),
+  }
+}
+
+export const getInstallations = async () => {
+    logger.info('getInstallations')
+
+    const nonInstallationRequest = getNonInstallationApp()
+   
+    
+    if(nonInstallationRequest !== null){
+      const response = await nonInstallationRequest('GET /app/installations')
+      return response.data
+    }
+    //TODO: maybe throw if there's no installation
```

Comment:
Example of where initials come in. Also a space between // and comment is eslint best practice
```suggestion
    // TODO: (bl) maybe throw if there's no installation
```